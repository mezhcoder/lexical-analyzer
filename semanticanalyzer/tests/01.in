program Test; var x: integer; begin x := 2+3*4; end.