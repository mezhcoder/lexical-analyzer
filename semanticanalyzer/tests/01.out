program
├── ID
├── declarations
│   └── var_declaration
│       ├── ID
│       └── TYPE
├── subprogram_declarations
└── compound_statement
    └── statement
        └── assign
            ├── ID
            └── Expression
                ├── Term
                │   └── INTEGER
                ├── +
                └── Term
                    ├── INTEGER
                    ├── *
                    └── INTEGER
Symbol Table
| Type    | Scope   |
|:--------|:--------|
| integer | Test    |
Type Table
|         | Type   |
|:--------|:-------|
| integer | x      |